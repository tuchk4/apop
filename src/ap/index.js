import crop from './actions/splice';
import filter from './actions/filter';
import join from './actions/join';
import map from './actions/map';
import remove from './actions/remove';
import sort from './actions/sort';
import swap from './actions/swap';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };

const ap = Object.assign(apply, sourceActions);
const recursive = Object.assign(recursiveApply, sourceActions);

export default Object.assign(ap, { recursive });
