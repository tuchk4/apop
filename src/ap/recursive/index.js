import crop from '../actions/splice';
import filter from '../actions/filter';
import join from '../actions/join';
import map from '../actions/map';
import remove from '../actions/remove';
import sort from '../actions/sort';
import swap from '../actions/swap';

import { recursiveApply } from '../utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };
const recursiveActions = {};

Object.keys(sourceActions).forEach(action => {
  recursiveActions[action] = (...args) =>
    recursiveApply(sourceActions[action](...args));
});

const recursive = Object.assign(
  (...actions) => recursiveApply(...actions),
  recursiveActions
);

export default recursive;
