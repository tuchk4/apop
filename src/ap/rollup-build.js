// import { rollup } from 'rollup';
//
// import resolve from 'rollup-plugin-node-resolve';
// import uglify from 'rollup-plugin-uglify';
// import commonjs from 'rollup-plugin-commonjs';
// import babel from 'rollup-plugin-babel';
// import { minify } from 'uglify-es';
// import babelrc from 'babelrc-rollup';
//
// const babelConfig = {
//   presets: [['es2015', { loose: true, modules: false }, 'stage-0']],
//   plugins: ['external-helpers', 'transform-object-rest-spread'],
// };
//
// var rollUpConfig = {
//   format: 'umd',
//   sourceMap: 'inline',
//   plugins: [
//     resolve({
//       jsnext: true,
//       main: true,
//       browser: true,
//     }),
//     commonjs(),
//     babel(
//       babelrc({
//         addExternalHelpersPlugin: false,
//         config: babelConfig,
//         exclude: 'node_modules/**',
//       })
//     ),
//   ],
// };
//
// async function make(config) {
//   let bundle = await rollup({
//     entry: `./src/${config.moduleName}.js`,
//     plugins: config.plugins,
//   });
//
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.js`,
//     sourceMap: config.sourceMap,
//   });
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.min.js`,
//     sourceMap: config.sourceMap,
//     plugins: [...rollUpConfig.plugins, uglify({}, minify)],
//   });
//   console.log(`make ./${config.moduleName}.js and ./${config.moduleName}.min.js`);
// }
//
// (async function() {
//
//   const builds = ['rmkArray','recursive/rmkArray'];
//
//   builds.forEach(async moduleName => await make({...rollUpConfig, moduleName}));
//
// })();
