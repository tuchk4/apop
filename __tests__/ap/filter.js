import ap from '../../src/build/ap';
const before = [1, 2, 3, 4, 5, [6, 7, 8, 9, 10, [11, 12, 13, 14, 15]]];

test('ap.recursive.filter', () => {
  const formula = ap.recursive(ap.filter(num => num > 3 && num < 13));

  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('ap.filter', () => {
  const formula = ap.filter(num => num > 3 && num < 13);

  const result = formula(before);

  expect(result).toMatchSnapshot();
});
