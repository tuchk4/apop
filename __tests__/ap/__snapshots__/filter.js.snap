// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ap.filter 1`] = `
Array [
  4,
  5,
]
`;

exports[`ap.recursive.filter 1`] = `
Array [
  4,
  5,
  Array [
    6,
    7,
    8,
    9,
    10,
    Array [
      11,
      12,
    ],
  ],
]
`;
